app/controllers/users_controller.rb

```diff
@@ -67,8 +67,11 @@ def handle_ip
     # Prevent someone from gaming the site by referring themselves.
     # Presumably, users are doing this from the same device so block
     # their ip after their ip appears three times in the database.
+    # Using split.first in case client is behind proxy to get his ip.
+    # If app is deployed on standalone server use request.remote_ip to
+    # get client ip address.
 
-    address = request.env['HTTP_X_FORWARDED_FOR']
+    address = request.env['HTTP_X_FORWARDED_FOR'].try(:split).try(:first) || request.remote_ip
```

Comment:
I think we need to be careful about 'split' here. X-Forwarded-For typically uses comma-space separation
(http://docs.aws.amazon.com/ElasticLoadBalancing/latest/DeveloperGuide/x-forwarded-headers.html) so we would need to strip the trailing commas, but it's not clear that X-Forwaded-For is a strict format.

    address = request.env.fetch('HTTP_X_FORWARDED_FOR', request.remote_ip).try(:split).try(:first).gsub(',', '')

But then I looked at the docs for `remote_ip` and that seems to actually be applying these proxy-aware rules already, so I think we should just do this:

    address = request.remote_ip

Unless I'm reading the docs wrong. http://apidock.com/rails/ActionDispatch/Request/remote_ip